src/com/github/igotyou/FactoryMod/Factorys/Compactor.java

```diff
@@ -0,0 +1,303 @@
+        return cp.getEnergyTime();
+    }
+    
+    public double getProductionTime() {
+        return cp.getProductionTime();
+    }
+    
+    public ItemList<NamedItemStack> getInputs() {
+        ItemList<NamedItemStack> inputs = new ItemList<NamedItemStack>();
+        Inventory inv = getInventory();
+        if(mode == CompactorMode.DECOMPACT) {
+            for(ItemStack is : inv.getContents()) {
+                if(is.getItemMeta().hasLore() && is.getItemMeta().getLore().contains(cp.getCompactLore())) {
+                    inputs.add(new NamedItemStack(is.getType(), is.getMaxStackSize(), is.getDurability(), is.getItemMeta().getDisplayName()));
+                    return inputs;
+                }
+            }
+            inputs.add(new NamedItemStack(Material.AIR, 1, (short)1, "Compacted Item"));
+        } else if (mode == CompactorMode.COMPACT) {
+            for(ItemStack is : inv.getContents()) {
+                if(is.getAmount() == is.getMaxStackSize()) {
+                    if(!is.getItemMeta().hasLore()) {
+                        inputs.add(new NamedItemStack(is.getType(), is.getAmount(), is.getDurability(), is.getItemMeta().getDisplayName()));
+                        inputs.addAll(cp.getRecipeMaterials());
+                        return inputs;
+                    }
+                }
+            }
+            return cp.getRecipeMaterials();
+        }
+        return inputs;
+    }
+    
+    public ItemList<NamedItemStack> getOutputs() {
+        ItemList<NamedItemStack> outputs = new ItemList<NamedItemStack>();
+        Inventory inv = getInventory();
+        if(mode == CompactorMode.DECOMPACT) {
+            for(ItemStack is : inv.getContents()) {
+                if(is.getItemMeta().hasLore() && is.getItemMeta().getLore().contains(cp.getCompactLore())) {
+                    outputs.add(new NamedItemStack(is.getType(), is.getMaxStackSize(), is.getDurability(), is.getItemMeta().getDisplayName()));
+                    return outputs;
+                }
+            }
+            outputs.add(new NamedItemStack(Material.AIR, 64, (short)1, "Decompacted Items"));
+        } else if (mode == CompactorMode.COMPACT) {
+            for(ItemStack is : inv.getContents()) {
+                if(is.getAmount() == is.getMaxStackSize()) {
+                    NamedItemStack compactedItemStack = new NamedItemStack(is.getType(), 1, is.getDurability(), is.getItemMeta().getDisplayName());
+                    compactedItemStack.getItemMeta().getLore().add(cp.getCompactLore());
+                    outputs.add(compactedItemStack);
+                    return outputs;
+                }
+            }
+            outputs.add(new NamedItemStack(Material.AIR, 1, (short)1, "Compacted Item"));
+        }
+        return outputs;
+    }
+    
+    public ItemList<NamedItemStack> getRepairs() {
+        return cp.getRepairMaterials();
+    }
+    
+    public void update() {
+        if(mode == CompactorMode.REPAIR) {
```

Comment:
I have done a review of Wild's first pass of refactoring. I also removed all duplicate factory manager code to an abstract class (AFactoryManager<T>). General cleanup and changes to proper implementation of generics across the plugin.
I also made all managers file agnostic. To make it much easier to swap out persistence for a database (an extension of wild's pass at a persistenceService implementation.
There still needs to be testing (honestly my inexperience with factory mod doesn't help me in this manner) along with reading existing persistence files.
I have merged repairFactory. And can merge compactor factory (gutting the code can get complicated)
On Jun 10, 2015 9:44 PM, Daniel Boston <notifications@github.com> wrote:In src/com/github/igotyou/FactoryMod/Factorys/Compactor.java:
> +                    compactedItemStack.getItemMeta().getLore().add(cp.getCompactLore());
> +                    outputs.add(compactedItemStack);
> +                    return outputs;
> +                }
> +            }
> +            outputs.add(new NamedItemStack(Material.AIR, 1, (short)1, "Compacted Item"));
> +        }
> +        return outputs;
> +    }
> +    
> +    public ItemList<NamedItemStack> getRepairs() {
> +        return cp.getRepairMaterials();
> +    }
> +    
> +    public void update() {
> +        if(mode == CompactorMode.REPAIR) {

What's the objection to fastracking the IO refactor? That seems way more important than the other pending merges.

—Reply to this email directly or view it on GitHub.